bindings/go/osvschema/vulnerability.go

```diff
@@ -0,0 +1,170 @@
+	Limit        string `json:"limit,omitempty"         yaml:"limit,omitempty"`
+}
+
+// Range describes the affected range of given version for a specific package.
+//
+// See: https://ossf.github.io/osv-schema/#affectedranges-field
+type Range struct {
+	Type             RangeType              `json:"type"                        yaml:"type"`
+	Events           []Event                `json:"events"                      yaml:"events"`
+	Repo             string                 `json:"repo,omitempty"              yaml:"repo,omitempty"`
+	DatabaseSpecific map[string]interface{} `json:"database_specific,omitempty" yaml:"database_specific,omitempty"`
+}
+
+// Severity is used to describe the severity of a vulnerability for an affected
+// package using one or more quantitative scoring methods.
+//
+// See: https://ossf.github.io/osv-schema/#severity-field
+type Severity struct {
+	Type  SeverityType `json:"type"  yaml:"type"`
+	Score string       `json:"score" yaml:"score"`
+}
+
+// Affected describes an affected package version, meaning one instance that
+// contains the vulnerability.
+//
+// See: https://ossf.github.io/osv-schema/#affected-fields
+type Affected struct {
+	Package           Package                `json:"package,omitempty"            yaml:"package,omitempty"`
+	Severity          []Severity             `json:"severity,omitempty"           yaml:"severity,omitempty"`
+	Ranges            []Range                `json:"ranges,omitempty"             yaml:"ranges,omitempty"`
+	Versions          []string               `json:"versions,omitempty"           yaml:"versions,omitempty"`
+	DatabaseSpecific  map[string]interface{} `json:"database_specific,omitempty"  yaml:"database_specific,omitempty"`
+	EcosystemSpecific map[string]interface{} `json:"ecosystem_specific,omitempty" yaml:"ecosystem_specific,omitempty"`
+}
+
+// MarshalJSON implements the json.Marshaler interface.
+//
+// This method ensures Package is only present if it is not equal to the zero value.
+// This is achieved by embedding the Affected struct with a pointer to Package used
+// to populate the "package" key in the JSON object.
+func (a Affected) MarshalJSON() ([]byte, error) {
+	type rawAffected Affected // alias Affected to avoid recursion during Marshal
+	type wrapper struct {
+		Package *Package `json:"package,omitempty"`
+		rawAffected
+	}
+	raw := wrapper{rawAffected: rawAffected(a)}
+	if a.Package == (Package{}) {
+		raw.Package = nil
+	} else {
+		raw.Package = &(a.Package)
+	}
+
+	return json.Marshal(raw)
+}
+
+// Reference links to additional information, advisories, issue tracker entries,
+// and so on about the vulnerability itself.
+//
+// See: https://ossf.github.io/osv-schema/#references-field
+type Reference struct {
+	Type ReferenceType `json:"type" yaml:"type"`
+	URL  string        `json:"url"  yaml:"url"`
+}
+
+// Credit gives credit for the discovery, confirmation, patch, or other events
```

Comment:
Standardize for consistency.

```suggestion
// Credit describes who to give credit to for the discovery, confirmation, patch, or other events
```